app/setup/index.js

```diff
@@ -0,0 +1,259 @@
+import fetch from 'isomorphic-fetch';
+import React, {Component} from 'react';
+import {render} from 'react-dom';
+import {baseUrl, isWebBrowser, dynamicRequireElectron} from '../utils/utils';
+
+const CLOUD = 'cloud';
+const ONPREM = 'onprem';
+const SERVER_TYPES = {
+    [CLOUD]: 'Plotly Cloud',
+    [ONPREM]: 'Plotly On-Premise'
+};
+
+class Setup extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            domain: '',
+            errorMessage: '',
+            loggedIn: false,
+            serverType: '',
+            status: '',
+            username: ''
+        };
+        this.authenticateUser = this.authenticateUser.bind(this);
+        this.buildOauthUrl = this.buildOauthUrl.bind(this);
+        this.chooseServerType = this.chooseServerType.bind(this);
+        this.setTimeoutAuthDone = this.setTimeoutAuthDone.bind(this);
+        this.updateStateWithEvent = this.updateStateWithEvent.bind(this);
+        this.verifyAuthDone = this.verifyAuthDone.bind(this);
+    }
+
+    buildOauthUrl() {
+        const oauthClientId = 'isFcew9naom2f1khSiMeAtzuOvHXHuLwhPsM7oPt';
+        const isOnPrem = this.state.serverType === ONPREM;
+        const plotlyDomain = isOnPrem ? this.state.domain : 'plot.ly';
+        const redirect_uri = baseUrl().replace('/setup', '');
+        return (
+            `https://${plotlyDomain}/o/authorize/?response_type=token&` +
+            `client_id=${oauthClientId}&` +
+            // TODO: Is there a better way? On premise has the `external-data-connector`
+            // route so cant use window.location.origin.
+            `redirect_uri=${redirect_uri}/oauth2/callback`
+        );
+    }
+
+    chooseServerType(serverType) {
+        this.setState({serverType});
+    }
+
+    verifyAuthDone() {
+        const {username} = this.state;
+        console.log(`baseUrl() ${baseUrl()}`);
+        fetch(`${baseUrl()}/hasauth`, {
+            method: 'POST',
+            headers: {
+                'Accept': 'application/json',
+                'Content-Type': 'application/json'
+            },
+            body: JSON.stringify({
+                username
+            })
+        }).then(res => res.json().then(json => {
+            if (res.status !== 200) {
+                this.setState({status: 'failure', errorMessage: json.error.message});
+            }
+            this.setState({loggedIn: json.hasAuth});
+        }));
+        // TODO: More error handling here? Probably need a catch here.
+    }
+
+    createCertificates() {
```

Comment:
should this be renamed? it seems weird that `createCertificates` just calls the `hasuath` endpoint (which doesn't create certificates, right?)